ansible_collections/arista/avd/roles/eos_designs/python_modules/inband_management/avdstructuredconfig.py

```diff
@@ -0,0 +1,221 @@
+                "ip_address": self._inband_management_ip,
+                "gateway": self._inband_management_gateway,
+                "type": "inband",
+            }
+        }
+
+    @cached_property
+    def static_routes(self) -> dict | None:
+        if self._inband_management_role != "child":
+            return None
+
+        return [
+            {
+                "destination_address_prefix": "0.0.0.0/0",
+                "gateway": self._inband_management_gateway,
+            }
+        ]
+
+    @cached_property
+    def _inband_management_data(self) -> dict:
+        vlans = []
+        subnets = []
+        peers = get(self._hostvars, f"avd_topology_peers..{self._hostname}", separator="..", default=[])
+        avd_switch_facts = get(self._hostvars, "avd_switch_facts", required=True)
+        for peer in peers:
+            if self._hostname not in get(avd_switch_facts, f"{peer}..switch..inband_management_parents", separator="..", default=[]):
+                continue
+
+            if (subnet := get(avd_switch_facts, f"{peer}..switch..inband_management_subnet", separator="..", required=True)) in subnets:
+                continue
+
+            subnets.append(subnet)
+            vlans.append(get(avd_switch_facts, f"{peer}..switch..inband_management_vlan", separator="..", required=True))
+
+        if not subnets:
+            return {}
+
+        return {
+            "vlans": vlans,
+            "subnets": subnets,
+        }
+
+    @cached_property
+    def vlan_interfaces(self) -> dict | None:
+        if self._inband_management_role != "parent":
+            return None
+
+        vlan_interfaces = {}
+        for index, subnet in enumerate(self._inband_management_data["subnets"]):
+            vlan = self._inband_management_data["vlans"][index]
+            vlan_interface_name = f"Vlan{vlan}"
+            vlan_interfaces[vlan_interface_name] = self._get_svi_cfg(subnet)
+
+        if vlan_interfaces:
+            return vlan_interfaces
+
+        return None
+
+    def _get_svi_cfg(self, subnet) -> dict:
+        subnet = ip_network(subnet, strict=False)
+        hosts = list(subnet.hosts())
+        prefix = str(subnet.prefixlen)
+
+        if self._mlag and self._mlag_role == "secondary":
+            ip_address = f"{str(hosts[2])}/{prefix}"
+        else:
+            ip_address = f"{str(hosts[1])}/{prefix}"
+
+        return {
+            "description": "L2LEAF_INBAND_MGMT",
+            "shutdown": False,
+            "mtu": self._p2p_uplinks_mtu,
+            "ip_address": ip_address,
+            "ip_virtual_router_addresses": [str(hosts[0])],
+            "ip_attached_host_route_export": {
+                "distance": 19,
+            },
+        }
+
+    @cached_property
+    def ip_virtual_router_mac_address(self) -> str | None:
+        if self._inband_management_role != "parent":
+            return None
+
+        return str(get(self._hostvars, "switch.virtual_router_mac_address", required=True)).lower()
```

Comment:
Yes because the old check was only there to prevent reordering of structured config. Now we don't really care, so it won't hurt to set the same variable (ip_virtual_router_address) twice to the same value. So to remove dependencies on other logic, I made this configure what it needs.